src/main/java/edu/tseidler/model/Board.java

```diff
@@ -4,9 +4,9 @@
 
 public class Board {
```

Comment:
I think your Board class has too many functionalities to be named just "Board", I think there should be a new class something like "MaintainBoard", consider it ;  )